### PR TITLE
core: allow DEBUG in mutex.c to run without DEVELHELP

### PR DESCRIPTION
**Pull request by authmillenon**
Tried to debug something without DEVELHELP and found that `mutex.c` isn't able to run without it. So I replaced thread names with PIDs.

### CODE REVIEW

**Comment by authmillenon** [comment]
Could go into this release IMHO (since it is more a bug fix than a feature).

**Comment by OlegHahm** [comment]
Makes sense.

**Comment by authmillenon** [comment]
Murdock is happy, bit an ACK is missing.

**Comment by authmillenon** [comment]
(two ACKs?)

File: core/mutex.c [hunk 1]
```diff
@@ -40,18 +40,18 @@
 int _mutex_lock(mutex_t *mutex, int blocking)
 {
     unsigned irqstate = irq_disable();
-    DEBUG("%s: Mutex in use.\n", sched_active_thread->name);
+    DEBUG("PID[%u]: Mutex in use.\n", sched_active_thread->pid);
```
Comment (Yonezawa-T2) [inline core/mutex.c]: `PRIkernel_pid`?

**Comment by authmillenon** [comment]
Addressed comment and unified a few other things.

**Comment by haukepetersen** [comment]
ACK, please squash

**Comment by haukepetersen** [comment]
@authmillenon: please squash.

@kaspar030, @OlegHahm: one of you up for a 2nd ack?

File: core/mutex.c [hunk 2]
```diff
@@ -107,7 +107,7 @@ void mutex_unlock(mutex_t *mutex)
 
 void mutex_unlock_and_sleep(mutex_t *mutex)
 {
-    DEBUG("%s: unlocking mutex. queue.next: 0x%08x pid: %" PRIkernel_pid ", and taking a nap\n", sched_active_thread->name, (unsigned)mutex->queue.next, sched_active_pid);
+    DEBUG("PID[%" PRIkernel_pid "]: unlocking mutex. queue.next: 0x%08x, and taking a nap\n", sched_active_pid, (unsigned)mutex->queue.next);
```
Comment (OlegHahm) [inline core/mutex.c]: While you're at it, you could fix the length of this line.

Comment (authmillenon) [inline core/mutex.c]: Done.

**Comment by authmillenon** [comment]
Squashed and addressed comments. While I was at it, I piggy-backed some code-style fixes

File: core/mutex.c [hunk 3]
```diff
@@ -35,26 +35,27 @@
...
-        DEBUG("%s: mutex_wait early out.\n", sched_active_thread->name);
+        DEBUG("PID[%" PRIkernel_pid "]: mutex_wait early out.\n", sched_active_pid);
         irq_restore(irqstate);
         return 1;
     }
     else if (blocking) {
-        thread_t *me = (thread_t*) sched_active_thread;
-        DEBUG("%s: Adding node to mutex queue: prio: %" PRIu32 "\n", me->name, (uint32_t)me->priority);
+        thread_t *me = (thread_t *)sched_active_thread;
+        DEBUG("PID[%" PRIkernel_pid "]: Adding node to mutex queue: prio: %" PRIu32 "\n", sched_active_pid, (uint32_t)me->priority);
```
Comment (haukepetersen) [inline core/mutex.c]: if you fix the line length with this PR, you might as well fix it for all DEBUG statements?!

**Comment by haukepetersen** [comment]
re-ACK if the one comment is adderssed. 

@OlegHahm @kaspar030 ping

**Comment by authmillenon** [comment]
Addressed comment and squashed immediately.

File: core/mutex.c [hunk 4]
```diff
@@ -35,34 +35,38 @@
 #define ENABLE_DEBUG    (0)
 #include "debug.h"
 
-#define MUTEX_LOCKED ((void*)-1)
+#define MUTEX_LOCKED ((void *)-1)
```
Comment (OlegHahm) [inline core/mutex.c]: unrelated (and unnecessary)?

**Comment by OlegHahm** [comment]
Apart from the - IMO superfluous - white spaces: ACK

**Comment by kaspar030** [comment]
> I piggy-backed some code-style fixes

As they are debatable, please remove those.

**Comment by authmillenon** [comment]
@kaspar030 where is the debate? I just see encouragements to shorten the lines....

File: core/mutex.c (hunk 4 above)
Comment (authmillenon) [inline core/mutex.c]: Removed, but don't agree... Pointers should be spaced according to the CC, so why not their casts?

**Comment by authmillenon** [comment]
Uglified casts ;-), squashed immediately.

**Comment by haukepetersen** [comment]
re-ACK

**Comment by haukepetersen** [comment]
CI is green -> go

**Comment by haukepetersen** [comment]
backport needed

**Comment by authmillenon** [comment]
Murdock builds it again (I think). So I remove the label.

**Pull request merged**